src/features/profile/ProfileFriends.ts

```diff
@@ -44,6 +54,57 @@ export class ProfileFriendsFormula extends PackItem {
 			}
 		});
 
+		pack.addSyncTable({
+			name: this.name(),
+			identityName: 'Friend',
+			schema: ProfileFriendsSchema,
+			connectionRequirement: coda.ConnectionRequirement.Required,
+			formula: {
+				name: 'FetchFriends',
+				description: 'Gets all friends',
+				parameters: [
+					profileIdParameter()
+				],
+				execute: async ([profileId], context) => {
+					profileId = cleanId(profileId);
+	
+					try {
+						let data = await context.fetcher.fetch({
```

Comment:
You could create a function that GET the list of friends and returns it, to avoid copying the same code twice (formula + table)

Reply:
Ooo you're right. I'm not actually much of a coder, so I don't always know the most "direct" option... just enough to get me in trouble 😆 
Would you prefer the function be defined in a separate file (e,g, the utility functions) or just defined within the ProfileFriends.ts?

Reply:
No worries😄For now you can keep the function in the same file, we don't need it anywhere. We'll move it later if necessary

Reply:
Okay so I'm not entirely sure how to define the function and then call it later.
![image](https://github.com/user-attachments/assets/bbaa5bfd-3907-47cd-b2e6-0786cef8fa0f)
I'm getting an error that it can't find `profileId` (thinking that it's because the function is declared outside of the class, but honestly no idea.)
Further, getting an issue where it's saying the type is wrong.
![image](https://github.com/user-attachments/assets/9782f548-aa49-43f6-afc3-a3b9d9d5461a)

What am I doing wrong here?

Reply:
You can add the function directly to the class : 
```ts
export class ProfileFriendsFormula extends PackItem {
  private async fetchFriends(id: string, context: coda.ExecutionContext) { 
    // code 
  }
}
```

and then call it in the execute like this :
```ts
execute: async ([profileId], context) => {
  return await this.fetchFriends(profileId, context);
}
```

`profileId` was not found in your code because the variable is available as an argument of the execute function (and you didn't get any argument from the execute)